### DISCUSSION TITLE
Massive RAM consumption on Mac M1, when running PowerShell script with Azure cmdlets.

### DISCUSSION THREAD

**Discussion by BorysMariusz** [PowerShell/PowerShell#19370] [General]
Hello
I am working on the latest version of macos and vscode and powershell core. When I am using local script to operat on Azure Cloud resources pwsh consume big amount of RAM.

<img width="754" alt="image" src="https://user-images.githubusercontent.com/32832364/226393728-e93e9dcb-eaf9-4825-b032-fde622b8f7eb.png">

**Comment by BorysMariusz** [comment]
Can anyone from @PowerShellTeam  help me with that ?

**Reply by StevenBucher98** [reply]
I suggest maybe posting this problem in the Issues tab of the PowerShell/PowerShell repo or in the https://github.com/Azure/azure-powershell repo if you believe it has something to do with the cmdlets from that module.

**Comment by rhubarb-geek-nz** [comment]
There is a configuration file you can pass to `pwsh` which should let you configure limits on RAM usage

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_session_configuration_files?view=powershell-7.3

This one looks promising

```
MaxMemoryPerShellMB 
```

In terms of mechanics, I hope this governs the garbage collector.

**Reply by BorysMariusz** [reply]
I think it is only applicable for Windows environments, and for session not terminals.

**Comment by BorysMariusz** [comment]
I think I have the solution @andschwa gave me the option which for now, fixed the ram usage. In Vscode options I have enabled powershell.analyzeOpenDocumentsOnly

**Reply by andyleejordan** [reply]
Glad it's working as expected now! Just linking the issue for posterity: https://github.com/PowerShell/vscode-powershell/issues/4496
